Dynmap module

Martmists-GH opened the pull request:
All TODO() calls need fixing before being merged.

Martmists-GH committed:
Initial dynmap implementation

Martmists-GH committed:
Improve player methods

Martmists-GH committed:
Format licenses

Martmists-GH committed:
Improvements

Martmists-GH committed:
Welcome to idiot town, population: Me

i509VCB commented on essentials-module-dynmap/src/main/java/io/github/nyliummc/essentials/mixin/dynmap/DynmapCoreMixin.java:
```diff
@@ -0,0 +1,25 @@
+class DynmapCoreMixin {
+    @Redirect(method="initConfiguration", at=@At(value="NEW", target="java.io.File", ordinal=0))
+    File configFile() throws IOException {
+        File f = new File(FabricLoader.getInstance().getConfigDirectory().getCanonicalPath() + "/essentials-dynmap.yaml");
```
Should we making an essentials folder?

We seem to be annexing most of the root server folder.

Martmists-GH replied:
Good idea, but out of scope for this PR I think

i509VCB commented on essentials-module-dynmap/src/main/kotlin/io/github/nyliummc/essentials/EssentialsDynmapModule.kt:
```diff
@@ -0,0 +1,91 @@
+    private fun initBlocks() {
+        var id = 0
+        var last: DynmapBlockState? = null
+        for (state in Block.STATE_IDS) {
```
Could this cause issues with registry sync?

i509VCB commented on essentials-module-dynmap/src/main/kotlin/io/github/nyliummc/essentials/entities/FabricDynmapOnlinePlayer.kt:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    override fun getArmorPoints(): Int {
+        return player.armor
```
I assume this refers to the `getArmor` method on LivingEntity that compensates for entity attributes and modifiers?

Martmists-GH replied:
Correct

i509VCB commented on essentials-module-dynmap/src/main/kotlin/io/github/nyliummc/essentials/entities/FabricDynmapServer.kt:
```diff
@@ -0,0 +1,156 @@
+        System.out.printf("Task requested with delay %d%n", delay)
+    }
+
+    override fun <T> callSyncMethod(task: Callable<T>?): Future<T>? {
```
There is an `executeAndJoin` or something like that method in MinecraftServer

i509VCB commented on essentials-module-dynmap/src/main/kotlin/io/github/nyliummc/essentials/entities/FabricDynmapServer.kt:
```diff
@@ -0,0 +1,156 @@
+    }
+
+    override fun reload() {
+        TODO("Not yet implemented")
```
What does this correspond on on other implementions?

Martmists-GH replied:
A full plugin reload on most, unsure about the details.

i509VCB replied:
If there is a sponge impl of dynmap, take a look at what they do there

i509VCB commented on essentials-module-dynmap/src/main/kotlin/io/github/nyliummc/essentials/entities/FabricDynmapServer.kt:
```diff
@@ -0,0 +1,156 @@
+
+    override fun getBiomeIDs(): Array<String?>? {
+        // TODO: Support translation keys if needed
+        return Registry.BIOME.ids.map(Identifier::toString).toTypedArray()
```
Registry sync:

A sponge story

Martmists-GH replied:
Any suggestions on how we can solve registry sync related issues? Though I don't think it'll be a big deal since it's a server-side mod.

i509VCB commented on essentials-module-dynmap/src/main/kotlin/io/github/nyliummc/essentials/entities/FabricDynmapWorld.kt:
```diff
@@ -0,0 +1,91 @@
+
+    override fun canGetSkyLightLevel(): Boolean {
+        // Always return true for now; It'll return 0 if unable
+        return true
```
I think this is dependent on `world.dimension`

Martmists-GH replied:
Yeah, looking into how to get the world right now.

i509VCB commented on essentials-module-dynmap/src/main/resources/essentials-dynmap.yaml:
```diff
@@ -0,0 +1,441 @@
+block-id-alias:
+#    "14": 1
+#    "15": 1
+#    "16": 1
```
If we have to tweak the config format to use identifiers, I am fine with you doing so

Martmists-GH replied:
This is just the default one for now, I'm looking into what all can be fixed/tweaked with mixins.

Martmists-GH committed:
License headers

Martmists-GH committed:
More TODO() calls for unimplemented methods

Martmists-GH committed:
World names are suffixes

Martmists-GH committed:
Merge branch 'master' into module-dynmap

# Conflicts:
#	build.gradle
#	essentials-api/src/main/kotlin/io/github/nyliummc/essentials/api/EssentialsModule.kt
#	gradle.properties

Martmists-GH committed:
:yert:

Martmists-GH committed:
Comment out and add default impls

Martmists-GH committed:
Merge branch 'master' into module-dynmap

# Conflicts:
#	essentials-api/src/main/kotlin/io/github/nyliummc/essentials/api/EssentialsMod.kt
#	gradle.properties
#	src/main/resources/fabric.mod.json

Martmists-GH committed:
Merge branch 'master' into module-dynmap

# Conflicts:
#	build.gradle
#	src/main/resources/fabric.mod.json

Martmists-GH committed:
Update gradle; Dependencies need to be moved and marked as 'shade'

Martmists-GH committed:
Thank you gradle very cool

Martmists-GH committed:
send help

Martmists-GH committed:
fix fabric.mod.json in fat jar

Martmists-GH committed:
Gradle struggles

Martmists-GH committed:
Fix mixin

Martmists-GH committed:
More dynmap

Martmists-GH committed:
license

Martmists-GH committed:
Finalize cf upload

Martmists-GH committed:
Dynmap is hell

Martmists-GH committed:
Update cf task

Martmists-GH committed:
A lot of dynmap progress

Martmists-GH committed:
idk what all I did but it was a lot

Martmists-GH committed:
I give up

Martmists-GH committed:
licenseformat

Martmists-GH committed:
a

Martmists-GH committed:
fix a gradle shit

Martmists-GH committed:
Start working on 1.16 version

Martmists-GH committed:
More fixes for 1.16, just some classloader issues left

Martmists-GH committed:
datapack stuff

Martmists-GH committed:
hecc

Martmists-GH committed:
More datapack stuff

Martmists-GH committed:
minifix

Martmists-GH committed:
gradle hell

Martmists-GH committed:
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

Martmists-GH committed:
Prepare for release

Martmists-GH committed:
minifix for cf task

i509VCB committed:
Actually describe what util module does

CamperSamu committed:
Dynmap version changed to 3.1-beta-1

DynmapCore-3.0-beta-10 isn't hosted anymore, updated DynmapCore to the latest beta version.

Martmists-GH committed:
Merge pull request #18 from CamperSamu/patch-1

Dynmap version changed to 3.1-beta-1

Martmists-GH committed:
Fix #17, #19 and cf issues

Martmists-GH committed:
licenseformat

Martmists-GH committed:
readme stuff
